File: jdollarx/src/main/java/info/testtools/jdollarx/custommatchers/IsPresent.java
package info.testtools.jdollarx.custommatchers;

import info.testtools.jdollarx.InBrowser;
import info.testtools.jdollarx.Path;
import org.hamcrest.Description;
import org.hamcrest.Matcher;
import org.hamcrest.TypeSafeMatcher;

public class IsPresent {

    public IsPresent(){}

    public static Matcher<Path> in(final InBrowser browser) {
        return new TypeSafeMatcher<Path>() {
            private Path el;

            @Override
            public void describeTo(final Description description) {
                description.appendText("browser page contains " + el.toString());
            }

            @Override
            protected void describeMismatchSafely(final Path el, final
            Description mismatchDescription) {
                mismatchDescription.appendText(CustomMatchersUtil.wrap(el) + " is absent");
            }

            @Override
            protected boolean matchesSafely(final Path el) {
                this.el = el;
                return browser.isPresent(el);
            }
        };
    }
}

File: dollarx/src/main/java/info/dollarx/custommatchers/scalatest/IsPresent.scala
package info.dollarx.custommatchers.scalatest

import info.dollarx.{Path, Browser}
import org.scalatest.matchers.{MatchResult, BeMatcher}


class IsPresent {
  def in(browser: Browser): BeMatcher[Path] = new BeMatcher[Path]{
    def apply(left: Path) =
      MatchResult(
        browser.isPresent(left),
        left.toString + " is expected to be present, but is absent",
        left.toString + " is expected to be absent, but is present"
      )
  }
}

